lib/Sema/CodeSynthesis.cpp

```diff
@@ -833,7 +833,7 @@ void TypeChecker::addImplicitConstructors(NominalTypeDecl *decl) {
 
   // Bail out if we're validating one of our constructors or stored properties
   // already; we'll revisit the issue later.
-  if (isa<ClassDecl>(decl)) {
+  if (isa<ClassDecl>(decl) && !decl->hasClangNode()) {
```

Comment:
We actually do run most of addImplicitInitializers() for Clang classes; we want to compute if convenience inits are inherited, even though nothing is synthesized. The specific block you’re looking at is skipped for imported classes because it calls getMembers() (and the check inside is always false).

I want to requestify addImplicitConstructors() at some point and split it up into several pieces but I haven’t quite figured out how I want it to look yet.